src/geometry_base/kd_tree_strategy.cc

```diff
@@ -0,0 +1,130 @@
+    const auto lane_position = lane->ToLanePosition(inertial_position);
+    if (lane_position.distance <= radius) {
+      api::RoadPositionResult road_position{
+          {lane, lane_position.lane_position}, lane_position.nearest_position, lane_position.distance};
+      road_positions.push_back(road_position);
+    }
+  }
+  return road_positions;
+}
+
+api::RoadPositionResult KDTreeStrategy::ClosestLane(const math::Vector3& point) const {
+  // Given that this method should actually obtain the lanes within a tolerance radius. Afterwards, the lane
+  // could be obtained using the point being closer to the centerline.
+  const api::InertialPosition& inertial_position{point.x(), point.y(), point.z()};
+  const MaliputPoint maliput_point = kd_tree_->Nearest(point);
+  const double radius = (point - maliput_point).norm() + 2 * sampling_step_;
+  const std::set<api::LaneId> lane_ids = ClosestLanes(point, radius);
+  const api::RoadGeometry* rg = GetRoadGeometry();
+
+  const api::Lane* lane_result = rg->ById().GetLane(maliput_point.lane_id().value());
+  api::LanePositionResult lane_position_result = lane_result->ToLanePosition(inertial_position);
+  api::RoadPositionResult road_position_result{{lane_result, lane_position_result.lane_position},
+                                               lane_position_result.nearest_position,
+                                               lane_position_result.distance};
+
+  for (const auto& current_lane_id : lane_ids) {
+    const api::Lane* current_lane = rg->ById().GetLane(current_lane_id);
+    const api::LanePositionResult current_lane_position = current_lane->ToLanePosition(inertial_position);
+    const api::RoadPositionResult current_road_position{{current_lane, current_lane_position.lane_position},
+                                                        current_lane_position.nearest_position,
+                                                        current_lane_position.distance};
+
+    if (IsNewRoadPositionResultCloser(current_road_position, road_position_result)) {
+      road_position_result = current_road_position;
+    }
+  }
+  return road_position_result;
+}
+
+std::set<api::LaneId> KDTreeStrategy::ClosestLanes(const math::Vector3& point, double distance) const {
+  const math::Vector3 min_corner{point.x() - distance, point.y() - distance, point.z() - distance};
+  const math::Vector3 max_corner{point.x() + distance, point.y() + distance, point.z() + distance};
+  const math::AxisAlignedBox search_region{min_corner, max_corner};
+  std::deque<const MaliputPoint*> maliput_points = kd_tree_->RangeSearch(search_region);
+  std::set<api::LaneId> maliput_lanes;
+  /*std::transform(maliput_points.begin(), maliput_points.end(),
+                   maliput_lanes.begin(),
+                   [](auto point) {return point->lane_id().value();});*/
+  for (const auto& current_point : maliput_points) {
+    maliput_lanes.insert(current_point->lane_id().value());
+  }
```

Comment:
Missing?

Reply:
It isn't possible as interestingly `std::set` only offers iterators that are constants: :disappointed: 

See https://en.cppreference.com/w/cpp/container/set/begin

>Because both iterator and const_iterator are constant iterators (and may in fact be the same type), it is not possible to mutate the elements of the container through an iterator returned by any of these member functions.